docs/api/app.md

```diff
@@ -1252,6 +1252,26 @@ Returns `String` - The command-line switch value.
 
 **Note:** When the switch is not present or has no value, it returns empty string.
 
+### `app.enableSecureMode()`
+
+Changes the default values of following `webPreferences`:
+- `sandbox: true`
+- `contextIsolation: true`
+- `nativeWindowOpen: true`
+- `enableRemoteModule: false`
+
+Removes the ability to use the following `webPreferences`:
+- `nodeIntegration`
+- `nodeIntegrationInWorker`
+
+When the following events are unhandled, `event.preventDefault()` is called implicitly:
+- `new-window`
+- `will-attach-webview`
+
+When no handler is set using `session.setPermissionRequestHandler()`, it behaves as if `callback(false)` was called.
+
+This method can only be called before app is ready.
```

Comment:
This will involve a race condition, is there any technical reason why this is the case.  I would imagine it doesn't matter when this is called?

Reply:
it probably does not matter, but it seems more future-proof to make this limitation? it should be enabled before any windows are created.

Reply:
Let's make that the restriction then instead of before app ready.  Enforce that it must be called before any webcontents are made.

Reply:
my concern is that people are used to the limitation that APIs must / cannot be called before / after `app.isReady`. do we have any API which can only be used before webcontents are created?

Reply:
Dumb question, would it make sense to make this a `webPreferences` argument instead? That would prevent the race condition and also resolve the question of when the API call was allowed.

Reply:
btw where is the race condition?

Reply:
@miniak It is very easy to _delay_ a task (run something after an event)
There is no easy way to guarantee to do something before an event.

If there is not a technical limitation here, I would suggest we tie this to the actual thing we care about not some other arbitrary event.

Reply:
Ok, I will change the condition. Is checking if there are 0 webcontents instances alive at the time the API is called good enough check?

Reply:
`webContents().getAll().length === 0` sounds like it will suffice for the actual condition you are trying to enforce here.  (That this API is called before any windows / web contents are made)

Reply:
@MarshallOfSound I've implemented the check in native code, can you check that? if you don't like the approach, I can do it in JS instead